indico/modules/events/contributions/controllers.py

```diff
@@ -373,3 +375,52 @@ class RHContributionsExportPDFBookSorted(RHManageContributionsActionsBase):
     def _process(self):
         pdf = ContributionBook(self._conf, session.user, self.contribs, tz=self.event_new.timezone, sort_by='boardNo')
         return send_file('book_of_abstracts.pdf', pdf.generate(), 'application/pdf')
+
+
+class RHManageContributionTypes(RHManageContributionsBase):
+    """Dialog to manage the ContributionTypes of an event"""
+
+    def _process(self):
+        contribution_types = self.event_new.contribution_types.all()
+        return jsonify_template('events/contributions/management/types_dialog.html', event=self.event_new,
+                                contribution_types=contribution_types)
+
+
+class RHManageContributionTypeBase(RHManageContributionsBase):
+
```

Comment:
normalize_url_spec = {
        'locators': {
            lambda self: self.contribution_type
        }
    }

Otherwise one could access a contribution type from a different event. The normalization code uses the locator to redirect to the proper page if needed